src/Plotting/plot_recipes.jl

```diff
@@ -161,6 +162,56 @@ julia> plot(B, 1e-2)  # faster but less accurate than the previous call
         seriesalpha --> DEFAULT_ALPHA
         seriescolor --> DEFAULT_COLOR
 
+        # extract limit and extrema of plotted shapes if existings
+        lims = Dict()
+        extr = Dict()
+        p = plotattributes[:plot_object]
+        if length(p) > 0
+            subplot = p[1]
+            for symbol in [:x, :y]
+                lims[symbol] = subplot[Symbol(symbol,:axis)][:lims]
+                extr[symbol] = subplot[Symbol(symbol,:axis)][:extrema]
+            end
+        end
+
+        if !isbounded(X)
+            # if the limit is :auto, set the limits to the current extrema
+            for symbol in [:x, :y]
+                if lims[symbol] == :auto
+                    emin = extr[symbol].emin
+                    emax = extr[symbol].emax
+                    # if the extrema is (-Inf,Inf), i.e. an empty plot,
+                    # set it to (0.,1.)
+                    lmin = isinf(emin) ? 0.0 : emin
```

Comment:
```suggestion
                    lmin = isinf(emin) ? -1.0 : emin
```
A default window centered in the origin feels slightly better (requires to update the comment above as well).

Reply:
> requires to update the comment above as well

Reply:
Sure, thanks. Was just not pushed already